arangod/Pregel/Worker/Worker.cpp

```diff
@@ -787,7 +787,9 @@ auto Worker<V, E, M>::loadGraph(LoadGraph const& graph)
 
   LOG_PREGEL("52070", WARN) << fmt::format(
       "Worker for execution number {} is loading", _config.executionNumber());
-  return std::move(_graphStore->loadShards(&_config, _makeStatusCallback()))
+  return futures::makeFutureWith([&]() {
```

Comment:
Why not `futures::makeFuture(_graphStore->loadShards(&_config, _makeStatusCallback()))`?

Reply:
Thanks for the tipp, makeFuture is much better here :)

We decided that in Pregel we want to get rid of as many exceptions as possible and use ResultT instead. This makes it more obvious where errors could occur and force you to handle them directly.